android/src/app/organicmaps/location/LocationHelper.java

```diff
@@ -369,6 +371,41 @@ public void stop()
     mActive = false;
   }
 
+  /**
+   * Resume location services when entering the foreground.
+   */
+  public void resumeLocationInForeground()
+  {
+    if (isActive())
+      return;
+    else if (LocationState.nativeGetMode() == LocationState.NOT_FOLLOW_NO_POSITION)
+    {
+      Logger.i(LOCATION_TAG, "Location updates are stopped by the user manually.");
+      return;
+    }
+    else if (ContextCompat.checkSelfPermission(mContext, ACCESS_FINE_LOCATION) != PERMISSION_GRANTED &&
+             ContextCompat.checkSelfPermission(mContext, ACCESS_COARSE_LOCATION) != PERMISSION_GRANTED)
+    {
+      Logger.i(LOCATION_TAG, "Permissions ACCESS_FINE_LOCATION and ACCESS_COARSE_LOCATION are not granted");
+      return;
+    }
+
+    Logger.i(LOCATION_TAG);
+    start();
+  }
+
+  /**
+   * Pause location services when entering the background.
+   */
+  public void pauseLocationInBackground()
+  {
+    if (!isActive())
+      return;
+
+    Logger.i(LOCATION_TAG);
```

Comment:
Not needed, we already have verbose log in stop()
Also pause can call stop directly. Or remove pauseLocationInBackground and call stop() on a client side.

Reply:
> Or remove pauseLocationInBackground and call stop() on a client side.

ok. I plan to add checks for IsNavigating() && Android Auto, but probably MwmApplication is better place for that.